app/tests/src/ingestion/test_pdf_postprocess.py

```diff
@@ -341,3 +354,36 @@ def test__add_link_markdown(caplog):
             ],
         ),
     ]
+
+
+def test_add_list_markdown(enriched_texts):
+    enriched_texts.append(
+        EnrichedText(
+            text="• Sub nested item.",
+            type=TextType.LIST_ITEM,
+            headings=[Heading(title="Section 1", level=1)],
+            page_number=2,
+            stylings=[
+                Styling(
+                    text="• Sub nested item.",
+                    pageno=2,
+                    headings=[Heading(title="Section 1", level=1)],
+                    wider_text="Sub nested item.",
+                    bold=False,
+                )
+            ],
+        )
+    )
+
+    prev_enriched_text_val = None
+    markdown_text = []
+    for enriched_text in enriched_texts:
+        if prev_enriched_text_val is not None:
+            markdown_text.append(_add_list_markdown(prev_enriched_text_val, enriched_text))
+        prev_enriched_text_val = enriched_text
```

Comment:
Curious to see this much logic in a unit test. I understand you're trying to replicate the logic in `add_markdown()` so maybe just call `add_markdown()` with the intent of testing `_add_list_markdown()`? Alternatively, call `_add_list_markdown()` with different inputs:
```
_add_list_markdown(None, enriched_text)
assert ...

_add_list_markdown(prev_enriched_text, enriched_text)
assert ...

etc.
```